thingsboard_gateway/gateway/tb_gateway_service.py

```diff
@@ -634,23 +634,25 @@ def __send_to_storage(self):
 
                         max_data_size = self.__config["thingsboard"].get("maxPayloadSizeBytes", 1024)
                         if self.__get_data_size(data) >= max_data_size:
+                            # Data is too large, so we will attempt to send in pieces
                             adopted_data = {"deviceName": data['deviceName'],
                                             "deviceType": data['deviceType'],
                                             "attributes": {},
                                             "telemetry": []}
+
+                            # First, loop through the attributes
                             for attribute in data['attributes']:
+                                adopted_data['attributes'].update({attribute: data['attributes'][attribute]})
```

Comment:
`adopted_data['attributes'].update(attribute)`

`attribute` - dict object
`data['attributes']`is a list and you want to get the item from this list by dict!
Also, we want in this place simply update attributes in adopted data, so can update by attribute without extra manipulations.

Reply:
@samson0v I will be honest, I am having a slightly hard time understanding your feedback in the review here.

All I did was to move this line you have commented on from after the size check to before it.

Are you suggesting that the line which is currently:
```
adopted_data['attributes'].update({attribute: data['attributes'][attribute]})
```
needs to be changed to:
```
adopted_data['attributes'].update(attribute)
```
?

Reply:
Yes